packages/components/src/components/BaseInput/BaseInput.vue

```diff
@@ -8,188 +7,140 @@
-		 */
-		modelValue: {
-			required: false,
-			type: [String, Number]
-		},
-		/**
-		 * If true, will mark the field as red
-		 */
-		error: {
-			type: Boolean,
-			required: false,
-			default: false
-		},
-		/**
-		 * HTML input type (email, password, etc)
-		 */
-		type: {
-			type: String,
-			required: false,
-			default: 'text'
-		},
-		/**
-		 * Icon that appears at the end of the input
-		 */
-		icon: {
-			type: String,
-			required: false
-		},
-		/**
-		 * whether to show clear button
-		 */
-		clearable: {
-			type: Boolean,
-			required: false,
-			default: false
-		},
-		/**
-		 * Input size. Can be one of "narrow", "big"
-		 */
-		size: {
-			type: String,
-			required: false
-		},
-		fontFamily: {
-			type: String,
-			required: false
-		},
-		class: {
+};
+</script>
 
-		}
-	},
-	data () {
-		return {
-			localValue: this.modelValue || ''
-		}
-	},
+<script lang="ts" setup>
+import { ref, computed } from 'vue';
+interface IProps {
+	/**
+	 * v-model/modelValue for the input
+	 */
+	modelValue?: string | number;
 
-	computed: {
-		cssClass () {
-			return this.class
-		},
-		showClear () {
-			return this.clearable && this.localValue && this.localValue.length > 0
-		},
-		hasSuffixContent () {
-			return this.icon || this.showClear
-		},
-		inputValue: {
-			get () {
-				return this.modelValue !== 'undefined' ? this.modelValue : ''
-			},
-			set (newValue) {
-				/** Updates the input value for the v-model **/
-				this.$emit('update:modelValue', newValue)
-				this.localValue = newValue
-			}
-		},
-		getStyle () {
-			let style = {
-				fontFamily: this.fontFamily ? this.fontFamily : null
-			}
-			return style
-		}
-	},
-	methods: {
-		onKeyDown (e) {
-			if (e.shiftKey) {
-				e.stopPropagation()
-			}
-		},
-		focus () {
-			this.$refs.input.focus()
-		},
-		blur () {
-			this.$refs.input.blur()
-		}
+	/**
+	 * If true, will mark the field as red
+	 */
+	error?: boolean;
+	/**
+	 * HTML input type (email, password, etc)
+	 */
+	type?: string;
+	/**
+	 * Icon that appears at the end of the input
+	 */
+	icon?: string;
+	/**
+	 * whether to show clear button
+	 */
+	clearable?: boolean;
+	/**
+	 * Input size. Can be one of "narrow", "big"
+	 */
+	size?: string;
+
+	fontFamily?: string;
+}
+const props = withDefaults(defineProps<IProps>(), {
+	modelValue: '',
+	error: false,
+	name: 'BaseInput',
+	type: 'text',
+	clearable: false,
+});
+
+defineEmits<{
+	(e: 'update:modelValue', value: string | number): void;
+}>();
 
+const showClear = computed(() => {
+	return props.clearable && props.modelValue ? true : false;
+});
+
+const hasSuffixContent = computed(() => {
+	return props.icon || showClear.value;
+});
+
+const getStyle = computed(() => {
+	return {
+		fontFamily: props.fontFamily,
+	};
+});
+
+function onKeyDown(e: KeyboardEvent) {
+	if (e.shiftKey) {
+		e.stopPropagation();
 	}
+}
 
+// Template Ref
+const input = ref<HTMLInputElement | null>(null);
+
+// @TODO Remove unused functions
```

Comment:
This is actually used by external components. It passes the HTMLElement.focus() function back to the actual component. When using ref on a component, you get the component instance instead of the $root element as is the case for native dom elements. The focus function is a shortcut to $baseInput.input.focus(). 

It can be removed, however, we would need to find the usages before removing it